File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        ArrayList<Integer> list = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            list.add(temp.val);
            temp=temp.next;
        }
        ListNode dummy = new ListNode(0);
        ListNode dummy1 = dummy;
        int target = list.size()/2;
        for(int i=0; i<list.size(); i++){
            int v = list.get(i);
            if(i==target){
                continue;
            }
            else{
                dummy1.next = new ListNode(v);
                dummy1 = dummy1.next;

            }
        }
        return dummy.next;
        
    }
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        ans = []
        temp = head
        while temp != None:
            ans.append(temp.val)
            temp = temp.next
        tar = len(ans)/2

        answer = []
        for i in range(len(ans)):
            if i==tar:
                continue
            else:
                answer.append(ans[i])
        dummy = ListNode(0)
        dummy1 = dummy
        for a in answer:
            dummy1.next = ListNode(a)
            dummy1 = dummy1.next
        return dummy.next

        